### ISSUE TITLE
Improve matching query term policy

### ISSUE THREAD

**Issue by curquiza** [meilisearch/meilisearch#2578]
This improvement motivation is about letting the user customize the behavior of Meilisearch when retrieving documents according to the query words.

Related product discussion
- https://github.com/meilisearch/product/discussions/255

## Process

💫 This issue will be completed with links and all the necessary information all along the feature creation process

- [x] Create PRD, see [this file](https://www.notion.so/meilisearch/Matching-Query-Terms-Policy-Words-Ranking-Rules-5584f606f1384de3885afabc18358979) (sorry Meili internal only)
- [x] Create a prototype by providing a docker image -> see this [discussion](https://github.com/meilisearch/meilisearch/discussions/2639)
- [x] Apply changes according to feedback
- [X] Create the corresponding specification and link it to this issue -> https://github.com/meilisearch/specifications/pull/173

**Comment by curquiza** [comment]
One precision: if the discovery phase (prototype creation + user feedback) is not satisfying, we will postpone this feature addition

**Comment by curquiza** [comment]
The prototype is now available, more details [here](https://github.com/meilisearch/meilisearch/discussions/2639), please, feel free to give your feedback

**Comment by curquiza** [comment]
The spec is available here: https://github.com/meilisearch/specifications/pull/162

**Comment by curquiza** [comment]
I've added the spec (https://github.com/meilisearch/specifications/pull/173) and the finally dediced API changes

**Comment by curquiza** [comment]
Hello everyone watching this issue!

We have just released [v0.29.0rc1](https://github.com/meilisearch/meilisearch/releases/tag/v0.29.0rc1), which is a release candidate of v0.29.0 🔥 
It means you can test the just-introduced `matchingStrategy` parameter during the search! Binaries are attached to [the release](https://github.com/meilisearch/meilisearch/releases/tag/v0.29.0rc1), or you can you the docker image:
```bash
docker run -it --rm \
    -p 7700:7700 \
    getmeili/meilisearch:v0.29.0rc0
```

Let us know about any bugs or feedback! 😄 It would be really helpful.

FYI, the official v0.29.0 release will be available on 3rd October.
